### ISSUE TITLE
[Bug]: 最新版不兼容redir-host

### ISSUE THREAD

**Issue by hoyien** [Fndroid/clash_for_windows_pkg#4062]
### 请认真检查以下清单中的每一项

- [ ] 已经搜索过，没有发现类似issue
- [ ] 已经搜索过文档，没有发现相关内容
- [ ] 已经尝试使用过最新版，问题依旧存在
- [X] 使用的是官方版本（未替换及修改过安装目录程序文件）

### 软件版本

v0.20.17

### 操作系统

Windows x64

### 系统版本

19042.844

### 问题描述

最新版v0.20.17本CFW不兼容enhanced-mode: redir-host，只能fake-ip

### 复现步骤

yaml配置enhanced-mode: redir-host

### 日志文件

<details><summary>logs</summary><p>

```
进入Home Directory/logs文件夹，找到日志文件，将文件拖动至输入框或点击界面左下角小字，将打开文件内容粘贴替换这行
```

</p></details>

**Comment by rinknife** [comment]
我的也是这个问题，导入订阅后报错invalid mode:redir-host

**Comment by monbed** [comment]
上游已不再支持redir-host

https://github.com/Dreamacro/clash/releases/tag/premium

**Comment by qianyuezhan** [comment]
> 上游已不再支持redir-host
> 
> https://github.com/Dreamacro/clash/releases/tag/premium

请问下怎么解决呢

**Comment by sforman2861597** [comment]
请问下怎么解决呢

**Comment by EddyLee88** [comment]
对, v0.20.17已经不支持redir-host了, 需要使用fake-ip, 应该联系供应商将你们的订阅配置中的dns.enhanced-mode修改为fake-ip;
当然, 也可以使用CFW的预处理配置功能:
```
parsers:
  - url: https://此处替换为你的订阅地址
    yaml:
      commands:
        - dns.enhanced-mode=fake-ip
```
使用以上代码配置预处理配置, 即可每次更新订阅都覆盖enhanced-mode为fake-ip
更多用法可以参考官方文档[配置文件预处理](https://docs.cfw.lbyczf.com/contents/parser.html)
我也是研究了很久才搞定, 希望可以帮到你
